Use a consistent format when testing BigDecimal.to_s

zzak opened the pull request:
The previous examples were causing failures in CI, after ruby/bigdecimal#264 was merged.

For example:

```
Failure:
NumericExtFormattingTest#test_default_to_fs [/rails/activesupport/test/core_ext/numeric_ext_test.rb:446]:
--- expected
+++ actual
@@ -1 +1,3 @@
-"10000 10.0"
+# encoding: US-ASCII
+#    valid: true
+"10 00010.0"
```

The recommendation was to adjust the test to deal with the upstream change more flexibly.

zzak commented:
Ok, it seems like it's due to ruby/bigdecimal#264 👀

matthewd commented:
We can just switch to a number that formats consistently (`0.100001` would probably do?). We don't have an opinion on the actual output string, just that our code ends up hitting upstream's implementation. We definitely should not imply that upstream should reconsider fixing their bug just because we had a test (not real code) that depended on it.

Honestly I'm not sure `to_fs` should even have this behaviour at all... it was necessary when we were overriding `to_s`, but I don't see why anyone would want to call our separate method with that argument.

zzak committed:
Use a consistent format when testing BigDecimal.to_s

The previous examples were causing failures in CI, after ruby/bigdecimal#264 was merged.

For example:

```
Failure:
NumericExtFormattingTest#test_default_to_fs [/rails/activesupport/test/core_ext/numeric_ext_test.rb:446]:
--- expected
+++ actual
@@ -1 +1,3 @@
-"10000 10.0"
+# encoding: US-ASCII
+#    valid: true
+"10 00010.0"
```

The recommendation was to adjust the test to deal with the upstream change more flexibly.
